### PR TITLE
Create 0034-find-first-and-last-position-of-element-in-sorted-array.js

### PR DESCRIPTION
**Pull request by aadil42**
Solved find-first-and-last-position-of-element-in-sorted-array in js.

File(s) Added: 0034-find-first-and-last-position-of-element-in-sorted-array.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/find-first-and-last-position-of-element-in-sorted-array/submissions/985379429/

### CODE REVIEW

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        
+        const mid = Math.floor((left+right)/2);
+
+        if(target > nums[mid]) {
+          left = mid+1;
+        }
+        if(target < nums[mid]) {
+          right = mid-1;
+        }
+        // this is the meat of the code
```
Comment (aakhtar3) [inline javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js:31]: Suggestion: Remove comment and creat descriptive variables or functions

```js
const isTarget = (...);
if (isTarget) { ... }
```

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * @return {number[]}
+ */
+var searchRange = function(nums, target) {
+    
+    const result = [];
+
+    result.push(binarySearch(true));
+    result.push(binarySearch(false));
+
+    function binarySearch(leftBias) {
```
Comment (aakhtar3) [inline javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js:16]: Suggestion: name booleans that prompt a question

```js
isLeftBias
```

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js (hunk 1 above)
Comment (aadil42) [reply javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js]: done.

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js (hunk 2 above)
Comment (aadil42) [reply javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js]: Just a kind reminder that I updated the code as suggested.

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * @return {number[]}
+ */
+var searchRange = function(nums, target) {
+    
+    const result = [];
+
+    result.push(binarySearch(true));
+    result.push(binarySearch(false));
+
+    function binarySearch(isLeftBias) {
```
Comment (aakhtar3) [inline javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js:16]: Avoid nested functions

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      let left = 0;
+      let right = nums.length - 1;
+      let index = -1;
+
+      while(left <= right) {
+        
+        const mid = Math.floor((left+right)/2);
```
Comment (aakhtar3) [inline javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js:23]: Use bitwise

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js (hunk 4 above)
Comment (aadil42) [reply javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js]: done

File: javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js (hunk 3 above)
Comment (aadil42) [reply javascript/0034-find-first-and-last-position-of-element-in-sorted-array.js]: done.

**Review by aakhtar3 — APPROVED** [review]

**Pull request merged**